deploy/kurl/kotsadm/template/base/install.sh

```diff
@@ -499,6 +499,62 @@ function kotsadm_health_check() {
     return 0
 }
 
+# check statefulset readiness
+function check_stateful_set_readiness() {
+    statefulsetName=$1
+    statefulSetPodVersionLabel="controller-revision-hash"
+
+    # get the list of pods for the statefulset
+    pods=$(kubectl get pods -l "app=$statefulsetName" -o jsonpath="{.items[*].metadata.name}")
+
+    # validate if each pod has the same version label, owner reference and is running
+    currentVersion=""
```

Comment:
that check might be worth changing since there is the potential of there being multiple kotsadm replicas being deployed into an existing deployment, and the situation mentioned in [this comment](https://github.com/replicatedhq/kots/pull/3743/files#r1139086695) situation could apply